featuretools/tests/primitive_tests/test_direct_features.py

```diff
@@ -52,6 +54,43 @@ def test_direct_from_variable(es):
     assert v == [0, 1]
 
 
+def test_rename(es):
+    feat = ft.Feature(es['log']['id'], parent_entity=es['sessions'], primitive=Count)
+    feat.get_feature_names()
+    copy_feat = feat.rename("session_test")
+    assert feat.unique_name() != copy_feat.unique_name()
+    assert feat.get_name() != copy_feat.get_name()
+    assert feat.base_features[0].generate_name() == copy_feat.base_features[0].generate_name()
+    assert feat.entity == copy_feat.entity
+    assert feat._names != copy_feat._names
+    assert feat.get_feature_names() != copy_feat.get_feature_names()
+
+
+def test_rename_multioutput(es):
```

Comment:
I think we should talk about what the expected output behavior is for a renamed multi-output feature. It seems to be the case that most generate_names() functions for multi-output features do not depend on the feature's _name attribute, but rather on the base_feature's name.

In this example

`feat.get_name()` returns: `'N_MOST_COMMON(log.product_id, n=2)'`
`feat.get_feature_names()` returns: `['N_MOST_COMMON(log.product_id, n=2)[0]', 'N_MOST_COMMON(log.product_id, n=2)[1]']`

Even though `copy_feat.get_name()` returns: `session_test`
`copy_feat.get_feature_names()` returns: `['N_MOST_COMMON(log.product_id, n=2)[0]', 'N_MOST_COMMON(log.product_id, n=2)[1]']`

Is this the desired behavior or are we expecting it to return `session_test[0]', 'session_test[1]']`?

Reply:
we expect `'session_test[0]', 'session_test[1]'`